cost-analyzer/templates/kubecost-cluster-controller-template.yaml

```diff
@@ -294,7 +294,7 @@ spec:
           {{- else }}
           value: http://{{ $serviceName }}.{{ .Release.Namespace }}:{{ .Values.service.targetPort | default 9090 }}/model
           {{- end }}
-        {{- if .Values.clusterController.kubecostAPIKey }}
+        {{- if or .Values.clusterController.kubecostAPIKey (eq .Values.clusterController.createClusterControllerSecret false) }}
```

Comment:
clusterController.kubecostAPIKeyCheck will check if the secret exists and has the correct key

Reply:
Sorry if I'm confused about this. Why do we have the `(eq .Values.clusterController.createClusterControllerSecret false)` check here?

Regardless of whether the secret was created by the user or the Helm chart, don't we need to extract the `kubecostAPIKey` from the secret and set it as an env var?

Reply:
fixed!